Adding dynamic development addin

emccorkle opened the pull request:
## Description

Adds a second JMP addin to the repository. After installing, this addin can be configured to point to JSL-Hamcrest source on disk. It will use that source for all its actions. This addin is designed to help ease development, testing, and reviewing of new features and bug fixes.

Closes #66.

## Checklist
- [x] **I am adding new or changing current functionality.**
  - [x] I have added note(s) to `CHANGELOG.md` as necessary.

I have tested the main addin and dev addin manually.

## Contributing

- [x] I have read and agree to the [Contributor Agreement](https://github.com/sassoftware/jsl-hamcrest/blob/master/ContributorAgreement.txt)

<!--- Replace your name and e-mail below -->
Signed-off-by: Evan McCorkle <Evan.McCorkle@jmp.com>

emccorkle committed:
Added instructions for bumping version.

emccorkle committed:
Define ut addin home and ut addin scratch.

Configures where addin source is located and where
addin can write preferences.

emccorkle committed:
Allow tests to run without addin installed.

emccorkle committed:
Added dev addin that dynamically looks at a repo.

JustinChilton commented on Tests/RunTestsStandalone.jsl:
```diff
@@ -1,6 +1,9 @@
 // SPDX-License-Identifier: Apache-2.0
 
+If(Is Empty(ut addin home), ut addin home = Convert File Path("../"));
+If(Is Empty(ut addin scratch), ut addin scratch = Convert File Path("%TEMP%"));
```
Could we change this to `Get Path Variable( "TEMP" )` to ensure it is cross-platform? I think even just `$TEMP` would work since we are only using functions that know about path variables.

JustinChilton commented on Source/DevAddin/addinLoad.jsl:
```diff
@@ -0,0 +1,16 @@
+	Stop();
+);
+
+ut addin scratch = Convert File Path("$ADDIN_HOME(com.jmp.jslhamcrestdev)");
```
Can we use `Get Default Directory()` instead? Makes it so you don't have to have the add-in installed to run it.

emccorkle replied:
After discussing, we are going to leave this as `Convert File Path` and assume the addin is installed. This entire file is then going to get run before every action instead of just once when JMP starts. This prevents problems when symbols are deleted.

JustinChilton committed:
Add button to select directory.

emccorkle committed:
Reinclude All when running tests or initializing.

This fits well with the purpose of the dev addin. Your
in-development matchers and reporters will get reincluded
every time you run your tests.

emccorkle committed:
Reinitialize ut addin home before every action.

emccorkle committed:
Refactored configuration script and storage names.

JustinChilton committed:
Merge branch 'master' into dev-addin

emccorkle committed:
Use path variable instead of %TEMP%.

JustinChilton commented on Source/DevAddin/InitializeHome.jsl:
```diff
@@ -6,7 +6,7 @@ Names Default To Here(0);
 ut addin scratch = Convert File Path("$ADDIN_HOME(com.jmp.jslhamcrestdev)");
 ut addin home = Try(
-	Load Text File(ut addin scratch || "/DevConfiguration.txt"),
+	Load Text File(ut addin scratch || "/HomeConfiguration.txt"),
```
Remove extra `/`. I am not sure if it would work with double slashes on both hosts.

JustinChilton committed:
Prompt to config addin home instead of modal.

JustinChilton committed:
Move Ctrl+F5 accelerator to menu item.

emccorkle committed:
Rename dev action to avoid name collision.
